### DISCUSSION TITLE
Automatically publish packages through CI/CD (Github Actions)

### DISCUSSION THREAD

**Discussion by RashadAkoodieRabbit** [onepub-dev/onepub#534] [Support]
Hi everyone. I am really trying to get my packages published to OnePub through Github Actions. It is obviously expecting some confirmation by either typing (Y/n) to confirm the publication but I already do my version checks/dart analyze myself before trying to publish the package so I really don't need the confirmation step and it is now failing on the publish step.

Any way to confirm through the CLI/ override the confirmation step etc?

**Comment by bsutton** [comment]
Dart pub lish --force

**Reply by RashadAkoodieRabbit** [reply]
Thank you, I think if possible this should be added to the page documentation as there is no instruction on how to publish via CI/CD as it is not currently possible without parsing the `--force` argument

**Comment by bsutton** [comment]
We have added the following to the doco:

When publishing a package the 'dart pub publish' command will prompt the user to confirm the act of publishing. When running a CI/CD action there is no user involved, so you need to use the --force switch:
dart pub publish --force

https://app.gitbook.com/o/ecaaU3OrP2aDt9qZRymk/s/JazaE7yAHIXZjE9ucSqu/~/changes/187/guides/ci-cd

Does this seem sufficient?
